File: object-oriented-programming/inheritance/python/boat_class.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0


class Boat:

    def __init__(self, given_name, given_length, given_capacity, given_berths, given_unit_cost):
        self.__name = given_name
        self.__length = given_length
        self.__capacity = given_capacity
        self.__berths = given_berths
        self.__unit_cost = given_unit_cost


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    name = "Sir David Attenborough"
    length = 128.9
    capacity = 88
    berths = 90
    cost = 200000000
    boaty = Boat(name, length, capacity, berths, cost)

File: object-oriented-programming/inheritance/c-sharp/boat_class.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;


namespace IsaacCodeSamples
{


    class Boat()
    {
        private string name;
        private float length;
        private int capacity;
        private int berths;
        private float unitCost;

        // Constructor method
        public Boat(string givenName, float givenLength, int givenCapacity, int givenBerths, float givenUnitCost)
        {
            name = givenName
            length = givenLength
            capacity = givenCapacity
            berths = givenBerths
            unitCost = givenUnitCost
        }
    }
    

    class Testing
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            Boat myBoat = new Boat();  // Instantiate a new boat object
        }
    }


}
